feat: AddKVCacheTransfer

Shunkangz opened the pull request:
Add fine-grained kv cache transfer measurement in disaggregated serving case. The use can set the env `TRTLLM_KVCACHE_TIME_OUTPUT_PATH` to dump the metrics.

Shunkangz commented:
/bot run

chuangz0 commented:
/bot run

juney-nvidia commented:
@Shunkangz pls add detailed description of this MR.

@chuangz0  @Shixiaowei02  pls help review this MR.

Thanks
June

kaiyux commented:
@Shunkangz The DCO checking is failed - please sign off the commit by adding `-s` when commit, so the command becomes `git commit -s -m "Msg"`

Commit:
Add updateKVCacheTransfer

Fix deserialize problem

Add TimeHelper

Fix bug

Rollback change

Update code

Roll back the mpi signal catch

Update .gitlab-ci.yml file

Remove files

Reformat

Change data type

Fix worldsize problem

Remove unrelated file

Fix bug

Fix race condition

Reformat

Reformat

Refactor code

Reformat

Disable fine grain profiling if not set

Signed-off-by: Shunkang <shunkangz@nvidia.com>
Signed-off-by: Shunkang <182541032+Shunkangz@users.noreply.github.co>

chuangz0 commented:
/bot run

niukuo commented:
[PR_Github #268](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/268/) [ run ] triggered by Bot

niukuo commented:
[PR_Github #268](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/268/) [ run ] completed with state `FAILURE`
[/LLM/main/L0_MergeRequest_PR pipeline #258](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/258/) completed with status: 'FAILURE'

Shixiaowei02 commented:
/bot run

niukuo commented:
[PR_Github #307](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/307/) [ run ] triggered by Bot

niukuo commented:
[PR_Github #307](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/307/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #292](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/292/) completed with status: 'FAILURE'

Shunkangz commented:
/bot run

niukuo commented:
[PR_Github #356](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/356/) [ run ] triggered by Bot

niukuo commented:
[PR_Github #356](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/356/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #325](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/325/) completed with status: 'FAILURE'

Shunkangz commented:
/bot run

niukuo commented:
[PR_Github #393](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/393/) [ run ] triggered by Bot

niukuo commented:
[PR_Github #393](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/393/) [ run ] completed with state `SUCCESS`
[/LLM/main/L0_MergeRequest_PR pipeline #349](https://nv/trt-llm-cicd/job/main/job/L0_MergeRequest_PR/349/) completed with status: 'SUCCESS'

Shixiaowei02 commented:
/bot reuse-pipeline

niukuo commented:
[PR_Github #441](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/441/) [ reuse-pipeline ] triggered by Bot

niukuo commented:
[PR_Github #441](https://nv/trt-llm-cicd/job/helpers/job/PR_Github/441/) [ reuse-pipeline ] completed with state `SUCCESS`
Reusing [PR_Github #393](https://prod.blsm.nvidia.com/sw-tensorrt-top-1/job/LLM/job/helpers/job/PR_Github/393/) for commit `4859ee3`
